src/kudu/consensus/consensus_meta.cc

```diff
@@ -173,9 +173,23 @@ const string& ConsensusMetadata::leader_uuid() const {
   return leader_uuid_;
 }
 
+const LastKnownLeader& ConsensusMetadata::last_known_leader() const {
+  DFAKE_SCOPED_RECURSIVE_LOCK(fake_lock_);
+  return last_known_leader_;
+}
+
 void ConsensusMetadata::set_leader_uuid(string uuid) {
   DFAKE_SCOPED_RECURSIVE_LOCK(fake_lock_);
   leader_uuid_ = std::move(uuid);
+
+  // Only update last_known_leader_ when the current node
+  // 1) has won a leader election (LEADER)
+  // 2) receives AppendEntries from a legitimate leader (FOLLOWER)
+  if (!uuid.empty()) {
+    DCHECK(pb_.has_current_term());
```

Comment:
I see lots of similar DCHECK's in consensus_meta.cc :-(

Reply:
Should I use CHECK instead?

Reply:
We should run some of our tests against DEBUG mode kuduraft so we get coverage of this stuff. Anyway I think this one should be safe because it appears in current_term()